ako-crd-operator/internal/controller/secret_controller.go

```diff
@@ -0,0 +1,299 @@
+			return fmt.Errorf("reconciler %s is already registered", name)
+		}
+	}
+
+	r.reconcilers = append(r.reconcilers, reconciler)
+	r.Logger.Infof("Registered reconciler: %s", name)
+	return nil
+}
+
+// NotifyReconcilers notifies all registered reconcilers of AVI client updates
+// Each reconciler gets a unique AVI client from the client pool
+func (r *SecretReconciler) NotifyReconcilers(ctx context.Context, aviClients *utils.AviRestClientPool) error {
+	r.reconcilersMutex.RLock()
+	defer r.reconcilersMutex.RUnlock()
+
+	if len(r.reconcilers) == 0 {
+		r.Logger.Info("No reconcilers registered")
+		return nil
+	}
+
+	// Ensure we have enough clients for all reconcilers
+	if len(aviClients.AviClient) < len(r.reconcilers) {
+		return fmt.Errorf("insufficient AVI clients: have %d, need %d", len(aviClients.AviClient), len(r.reconcilers))
+	}
+
+	var errors []error
+	for i, reconciler := range r.reconcilers {
+		name := reconciler.GetReconcilerName()
+		r.Logger.Infof("Notifying reconciler %s of AVI client update", name)
+
+		// Give each reconciler a unique AVI client
+		aviClient := session.NewAviSessionClient(aviClients.AviClient[i])
+
+		if err := reconciler.UpdateAviClient(aviClient); err != nil {
+			r.Logger.Errorf("Failed to update AVI client for reconciler %s: %v", name, err)
+			errors = append(errors, fmt.Errorf("reconciler %s: %w", name, err))
+		} else {
+			r.Logger.Infof("Successfully updated AVI client for reconciler %s", name)
+		}
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("failed to update %d reconcilers: %v", len(errors), errors)
+	}
+
+	return nil
+}
+
+// GetRegisteredReconcilers returns the list of registered reconciler names
+func (r *SecretReconciler) GetRegisteredReconcilers() []string {
+	r.reconcilersMutex.RLock()
+	defer r.reconcilersMutex.RUnlock()
+
+	names := make([]string, 0, len(r.reconcilers))
+	for _, reconciler := range r.reconcilers {
+		names = append(names, reconciler.GetReconcilerName())
+	}
+	return names
+}
+
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
+// Reconcile handles secret update events
+func (r *SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Logger.WithValues("secret", req.NamespacedName, "traceID", uuid.New().String())
+	ctx = utils.LoggerWithContext(ctx, log)
+	log.Debug("Reconciling AVI secret update")
+	defer log.Debug("Reconciled AVI secret")
+
+	// Only process avi-secret in the AKO namespace
+	if req.Name != AviSecretName || req.Namespace != utils.GetAKONamespace() {
+		log.Debug("Ignoring non-AVI secret")
+		return ctrl.Result{}, nil
+	}
+
+	// Get the secret to verify it exists and has valid data
+	secret := &corev1.Secret{}
+	if err := r.Get(ctx, req.NamespacedName, secret); err != nil {
+		if k8serror.IsNotFound(err) {
+			log.Info("AVI secret not found, skipping reconciliation")
+			return ctrl.Result{}, nil
```

Comment:
should we retry till avi secret is created?

Reply:
We will get an event for seret creation, retry not needed